### PR TITLE
Check if middleware group exists before adding middleware to it

### PR DESCRIPTION
**Pull request by slackernrrd**
A more robust way to register the `InterceptFlexibleAttributes` middleware, that works even if `NovaCoreServiceProvider` get loaded after this provider. Solves https://github.com/whitecube/nova-flexible-content/issues/99

### CODE REVIEW

File: src/FieldServiceProvider.php [hunk 1]
```diff
@@ -19,7 +19,16 @@ class FieldServiceProvider extends ServiceProvider
      */
     public function boot()
     {
-        $this->app['router']->pushMiddlewareToGroup('nova', InterceptFlexibleAttributes::class);
+        $router = $this->app['router'];
+        if ($router->hasMiddlewareGroup('nova')) {
+            $this->app['router']->pushMiddlewareToGroup( 'nova', InterceptFlexibleAttributes::class );
```
Comment (voidgraphics) [inline src/FieldServiceProvider.php]: Why not use the `$router` variable you created?

File: src/FieldServiceProvider.php [hunk 2]
```diff
@@ -19,7 +19,16 @@ class FieldServiceProvider extends ServiceProvider
...
+        }
+        elseif (! $this->app->configurationIsCached()) {
+            $this->app['config']->set('nova.middleware', array_merge(
+                $this->app['config']->get('nova.middleware', []),
```
Comment (voidgraphics) [inline src/FieldServiceProvider.php]: Could this be more concise? Correct me if I'm wrong but couldn't this be simply written as `config('nova.middleware')`?

File: src/FieldServiceProvider.php (hunk 1 above)
Comment (slackernrrd) [reply src/FieldServiceProvider.php]: That was my intention. I guess you shouldn't send pull requests just before going to bed...

File: src/FieldServiceProvider.php (hunk 2 above)
Comment (slackernrrd) [reply src/FieldServiceProvider.php]: True! I was inspired by `ServiceProvider::mergeConfigFrom()` which is why my code was a bit more verbose.

**Comment by voidgraphics** [comment]
Been a while, sorry about that! Thanks for the PR.

**Pull request merged**